dfdaemon/server.go

```diff
@@ -117,14 +117,17 @@ func NewFromConfig(cfg config.Properties) (*Server, error) {
 
 // Start runs dfdaemon's http server.
 func (s *Server) Start() error {
+	var err error
 	_ = proxy.WithDirectHandler(handler.New())(s.proxy)
 	s.server.Handler = s.proxy
 	if s.server.TLSConfig != nil {
 		logrus.Infof("start dfdaemon https server on %s", s.server.Addr)
+		err = s.server.ListenAndServeTLS("", "")
```

Comment:
We can directly return the error here, then the `err` variable will not be needed.

Reply:
The code check will fail if written this way

Reply:
> The code check will fail if written this way

What's the error message exactly? I'm guessing it complains about the `else` block, since if your return in the `if` block, `else` is not needed.

Reply:
It's not a structured program

Reply:
I'm not quite sure what you mean. But I guess this would be what you want:

```go
if s.server.TLSConfig != nil {
        logrus.Infof("start dfdaemon https server on %s", s.server.Addr)
        return s.server.ListenAndServeTLS("", "")
}
logrus.Infof("start dfdaemon http server on %s", s.server.Addr)
return s.server.ListenAndServe()
```

Reply:
This code is not easy to understand，I can revise it if you insist

Reply:
I can certainly understand your argument, since the symmetry of the `if {} else {}` statement kind of matches the actual logic of the code. I'll probably insist on using early returns to reduce nesting and avoiding unnecessary intermediate variables in some other cases. But since this function is very short, I think it's fine to keep your current implementation.